File: crates/corelink-cli/examples/quickstart_stats.rs
//! `quickstart_stats` — Probe service health/stats via `GET /api/health`.
//!
//! Customer concept: liveness + readiness probe. The response includes
//! version, region, and per-dependency status — wire this into your
//! synthetic monitor. Anonymous endpoint (no PAT required).
//!
//! Run:
//!
//! ```text
//! cargo run --example quickstart_stats -p corelink-cli
//! ```

#![forbid(unsafe_code)]
#![allow(
    clippy::print_stdout,
    reason = "examples emit human-readable narration to stdout"
)]

use anyhow::{Context, Result};
use std::env;

#[tokio::main]
async fn main() -> Result<()> {
    let api = env::var("CORELINK_API_URL")
        .unwrap_or_else(|_| "https://sandbox.corelink.dev".to_string());

    let resp = reqwest::Client::new()
        .get(format!("{api}/api/health"))
        .send()
        .await
        .context("GET /api/health")?;

    let status = resp.status();
    if !status.is_success() {
        let body = resp.text().await.unwrap_or_default();
        anyhow::bail!("server returned {status}: {body}");
    }
    let body: serde_json::Value = resp.json().await.context("decode body")?;
    println!("status={status}");
    println!("health={}", serde_json::to_string_pretty(&body)?);
    Ok(())
}

File: tools/sdks/go/examples/quickstart_stats.go
//go:build ignore

// quickstart_stats — Probe service health/stats via GET /api/health.
//
// Customer concept: liveness + readiness probe. Returns version,
// region, and per-dependency status. Anonymous endpoint.
//
// Run: go run examples/quickstart_stats.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"time"
)

func main() {
	api := os.Getenv("CORELINK_API_URL")
	if api == "" {
		api = "https://corelink-api.humangr.com"
	}
	req, err := http.NewRequest("GET", api+"/api/health", nil)
	if err != nil {
		fmt.Fprintf(os.Stderr, "build request: %v\n", err)
		os.Exit(1)
	}
	req.Header.Set("Accept", "application/json")
	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Fprintf(os.Stderr, "http: %v\n", err)
		os.Exit(1)
	}
	defer resp.Body.Close()
	raw, _ := io.ReadAll(resp.Body)
	if resp.StatusCode/100 != 2 {
		fmt.Fprintf(os.Stderr, "server returned %d: %s\n", resp.StatusCode, string(raw))
		os.Exit(1)
	}
	var pretty bytes.Buffer
	if err := json.Indent(&pretty, raw, "", "  "); err != nil {
		pretty.Write(raw)
	}
	fmt.Printf("status=%d\n%s\n", resp.StatusCode, pretty.String())
}
